src/GitLink/Linker.cs

```diff
@@ -41,22 +41,37 @@ public static bool Link(string pdbPath, LinkOptions options = default(LinkOption
             {
                 sourceFiles = pdb.GetFilesAndChecksums().Keys.ToList();
 
-                repositoryDirectory = GitDirFinder.TreeWalkForGitDir(Path.GetDirectoryName(sourceFiles.First()));
-                if (repositoryDirectory == null)
+                if (options.GitWorkingDirectory != null)
```

Comment:
I think null, "" or " " are all invalid values for a working directory. Therefore we should check for more than just null, or am I missing something?